src/frontend/apps/lti_site/components/App/AppInitializer/index.tsx

```diff
@@ -21,11 +21,21 @@ import React, {
   useMemo,
   useState,
 } from 'react';
+import { createJSONStorage } from 'zustand/middleware';
 
 import { useIsFeatureEnabled } from 'data/hooks/useIsFeatureEnabled';
+import { Maybe } from 'lib-common';
 
-export const AppInitializer = ({ children }: PropsWithChildren<{}>) => {
+interface AppInitializerProps {
+  jwt: Maybe<string>;
+  refresh_token: Maybe<string>;
+}
+
+export const AppInitializer = (
+  props: PropsWithChildren<AppInitializerProps>,
+) => {
   const [isAppInitialized, setIsAppInitialized] = useState(false);
+  const [isJwtInitialized, setIsJwtInitialized] = useState(false);
```

Comment:
Maybe you can use `isAppInitialized` instead, I am not sure what is doing `isAppInitialized` actually, because depends their dependencies the others `useEffect` are potentially called after `isAppInitialized` is setted to `true`.

Reply:
The idea is to call it last yes and while `isAppInitialized` is not `true`, nothing is render